File: C/Code/2700/2798 - 블랙잭.c
#include <stdio.h>

int main(n, m) {
	scanf("%d %d", &n, &m);
	int arr[100];
	for (int i = 0; i < n; i++)
		scanf("%d", arr + i);
	for (int i = 0; i < n - 1; i++)			// bubble sort
		for (int j = 0; j < n - i - 1; j++)
			if (arr[j] > arr[j + 1]) {
				int temp = arr[j];
				arr[j] = arr[j + 1];
				arr[j + 1] = temp;
			}
	int result = 0;
	for (int i = 0; i < n - 2; i++)			// brute force
		for (int j = i + 1; j < n - 1; j++)
			for (int k = j + 1; k < n; k++) {
				int temp = arr[i] + arr[j] + arr[k];	// sum of 3 cards
				if ((temp <= m) && (temp > result))		// condition
					result = temp;
			}
	printf("%d", result);
	return 0;
}

File: Python/Code/2700/2798 - 블랙잭.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))
result = 0
for i in range(len(arr) - 2):
	for j in range (i + 1, len(arr) - 1):
		for k in range(j + 1, len(arr)):
			if result < arr[i] + arr[j] + arr[k] <= m:
				result = arr[i] + arr[j] + arr[k]
			if result == m:
				print(result)
				exit()
print(result)